test/onnx/internal/fx/test_frontend_consistency.py

```diff
@@ -0,0 +1,157 @@
+
+    def forward(self, tensor_x: torch.Tensor):
+        tensor_x = self.fc0(tensor_x)
+        tensor_x = torch.sigmoid(tensor_x)
+        tensor_x = self.fc1(tensor_x)
+        tensor_x = torch.sigmoid(tensor_x)
+        tensor_x = self.fc2(tensor_x)
+        tensor_x = torch.sigmoid(tensor_x)
+        output = self.fc3(tensor_x)
+        return output
+
+
+def module_inputs_mlp_model(
+    module_info, device, dtype, requires_grad, training, **kwargs
+):
+    make_input = functools.partial(
+        common_modules.make_tensor,
+        device=device,
+        dtype=dtype,
+        requires_grad=requires_grad,
+    )
+    module_inputs = [
+        common_modules.ModuleInput(
+            constructor_input=common_modules.FunctionInput(),
+            forward_input=common_modules.FunctionInput(make_input((97, 8))),
+        ),
+    ]
+    return module_inputs
+
+
+MODULES_DB = [
+    common_modules.ModuleInfo(MLPModel, module_inputs_func=module_inputs_mlp_model)
+]
+
+
+class TestFxFrontendConsistency(common_utils.TestCase):
+    @classmethod
+    def create_test_base(cls, fx_frontend: _FrontendType):
+        """Returns the base test method for the given fx frontend."""
+
+        @common_modules.modules(
+            MODULES_DB, train_eval_mode=common_modules.TrainEvalMode.eval_only
+        )
+        def _output_match_base(
+            self: common_utils.TestCase,
+            device: str,
+            dtype: torch.dtype,
+            module_info: common_modules.ModuleInfo,
+            training: common_modules.TrainEvalMode,
+        ):
+            """Base test method for testing each fx frontend, used by instantiate_device_type_tests."""
+            # device is provided by instantiate_device_type_tests, but we only want to run in cpu.
+            assert device == "cpu"
+
+            module_cls = module_info.module_cls
+            module_inputs = module_info.module_inputs_func(
+                module_info,
+                device=device,
+                dtype=dtype,
+                requires_grad=False,
+                training=False,
+            )
+
+            for (i, module_input) in enumerate(module_inputs):
+                with self.subTest(
+                    module_name=module_cls.__name__,
+                    sample_num=i,
+                    fx_frontend=fx_frontend.name,
+                ):
+                    init_args, init_kwargs = (
+                        module_input.constructor_input.args,
+                        module_input.constructor_input.kwargs,
+                    )
+                    model = module_cls(*init_args, **init_kwargs)
+                    model.to(device).to(dtype)
+                    model.eval()
+
+                    args, kwargs = (
+                        module_input.forward_input.args,
+                        module_input.forward_input.kwargs,
+                    )
+
+                    if isinstance(fx_frontend, frontend.FxFrontendUnpackKwargs):
```

Comment:
This looks strange. I feel we should try to keep the original input schema in FX exporter instead of making special handling in this class.

Reply:
You are right, this is not good looking.

Root cause being the assumption of only positional args are allowed by fx graph infrastructures/utilities/usages from almost everywhere. 

This part needs more thoughts. When writing this PR I was leaning towards gradually removing the need for `FxFrontendUnpackKwargs` as we eventually keep original input schema everywhere. But now I think what we need is a builtin input formatter and validator. I want to leave this as TODO since it's a larger chunk of component, to not block the other PRs after this.